examples/fluids/problems/sgs_dd_model.c

```diff
@@ -14,24 +14,26 @@
 typedef struct {
   CeedElemRestriction  elem_restr_grid_aniso, elem_restr_sgs;
   CeedVector           grid_aniso_ceed;
-  CeedQFunctionContext sgsdd_qfctx;
-} *SgsDDModelSetupData;
+  CeedQFunctionContext sgsdd_qfctx, ifunction_qfctx;
```

Comment:
Tangentially related - if you have access to an operator, you have access to the QFunction and QFunctionContext. How much can you trim back these structs throughout the fluids code?

Reply:
I'm assuming this is more about `ifunction_qfctx` than `sgsdd_qfctx`. 

Well, the main complication is that the IFunction operator is a composite operator and not all the suboperators have the same qfunction context. I'm not sure if there's a way to grab a suboperator by name (though that could be easily coded up  for the library or for the fluid's example exclusively I suppose).

Note also that this struct lives only inside this file; it's primary job is to ease the verbosity of function parameters within this setup function.

Reply:
Ah, I got around this by tracking the index of the relevant suboperator. Overall I've been on a campaign to reduce how much data we drag around in structs in Ratel but haven't started waging war on structs here in the libCEED repo